### PR TITLE
Stun Baton Eyecandy | Glows and Sparks

### PR DESCRIPTION
**Pull request by KitsuneQueen**
## About The Pull Request
Adds a soft glow and sparks to the stun baton and stun prod. Video below:
https://github.com/tgstation/tgstation/assets/70779744/468351ef-0574-46b9-8d32-58cf40651f3a
## Why It's Good For The Game
metro-cop

Looks cool.
## Changelog
:cl:
add: In an attempt to stop the greytide, NanoTrasen has increased security's baton energy output. This has, through testing, done nothing but make the device spark more than it used to.
/:cl:

### CODE REVIEW

File: code/game/objects/items/melee/baton.dm [hunk 1]
```diff
@@ -427,6 +427,11 @@
 	on_stun_volume = 50
 	active = FALSE
 	context_living_rmb_active = "Harmful Stun"
+	light_range = 2
+	light_system = OVERLAY_LIGHT
+	light_on = FALSE
+	light_color = LIGHT_COLOR_ORANGE
```
Comment (LemonInTheDark) [inline code/game/objects/items/melee/baton.dm:433]: at default lighting power this ends up being remarkably colorless, is that what you want?

also, are you sure you want this to be an overlay light? The range makes it very oppressive, which might be nice but I think would be paired better with a weaker, more faint light (also consider a range of 1.5 for something slightly more pulled into the source).

If what you want is just for the baton to glow (and not illuminate anything else), we have an emissives system that lets you cut things out from the darkness. That might work well here depending on your goal

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]
I love shit like this, thank you!

File: code/game/objects/items/melee/baton.dm [hunk 2]
```diff
@@ -541,6 +546,8 @@
 		active = !active
 		balloon_alert(user, "turned [active ? "on" : "off"]")
 		playsound(src, SFX_SPARKS, 75, TRUE, -1)
+		toggle_light(user)
+		do_sparks(1, TRUE, src)
 	else
```
[the remark this answers is not in the document]
Comment (WinterDarkraven) [reply code/game/objects/items/melee/baton.dm:551]: Oops, sorry! Working on it right now.

File: code/game/objects/items/melee/baton.dm [hunk 3]
```diff
@@ -550,6 +557,11 @@
 	update_appearance()
 	add_fingerprint(user)
 
+/obj/item/melee/baton/security/proc/toggle_light(mob/user)
+	var/old_light_on = light_on
+	set_light_on(!light_on)
+	return light_on != old_light_on // If the value of light_on didn't change, return false. Otherwise true.
```
[the remark this answers is not in the document]
Comment (WinterDarkraven) [reply code/game/objects/items/melee/baton.dm:563]: Will correctly format, and in the future! Thanks!

File: code/game/objects/items/melee/baton.dm (hunk 1 above)
Comment (WinterDarkraven) [reply code/game/objects/items/melee/baton.dm:433]: I didn't want the colours to be too harsh. I'll set the range to 1.5, the goal is for it to be soft glow that can be ignored. It's meant to make things look pretty, not standout by itself.

**Comment by newfren** [comment]
does this ignite plasma?

**Comment by WinterDarkraven** [comment]
> does this ignite plasma?

I believe so. Will test.

**Comment by WinterDarkraven** [comment]
Requests done, removed the weird un-used thing. Made it so when you attack with the baton it also sparks.

https://github.com/tgstation/tgstation/assets/70779744/be286b4a-7f05-40af-a2a9-9360370ea1cf

File: code/game/objects/items/melee/baton.dm (hunk 2 above)
[the remark this answers is not in the document]
Comment (WinterDarkraven) [reply code/game/objects/items/melee/baton.dm:551]: Removed the weird value. I don't think it's needed(?)

**Comment by WinterDarkraven** [comment]
I also just realised the stun prod already sparks, and this makes it spark WAY more than it already did. However, it uses a sparker, which I never thought of. I don't know if I should remove the sparker from the stun prod and let it use the baton do_sparks stuff, or move the sparker to the regular baton. Uhhh.

**Comment by SomeoneYouProbablyKnow** [comment]
Is it possible to make sparks on the target and not yourself when attacking? That might look even cooler

**Comment by WinterDarkraven** [comment]
> light_on != old_light_on // If the value of light_on didn't change, return false. Otherwise true.

Pretty new to coding, but I could make it spark both 

> Is it possible to make sparks on the target and not yourself when attacking? That might look even cooler

Still pretty new to coding. I'll see if I can spark both the user and victim. Would be cool since you start shaking afterwards too!

**Comment by WinterDarkraven** [comment]
Forgot to turn off the baton light when it runs out of charge, now the baton no longer glows when out of power. Comment fixed, because I goofed it the first time. The stun prod is no longer extra sparky and annoying to the ears.

**Comment by Admiral-Gigaton** [comment]
time to make combine officer cosplay

**Comment by WinterDarkraven** [comment]
I forgot to add "light_power" to the stun baton because I thought it wouldn't be needed. This gives admins and people in the future to easily edit the stun baton's range, brightness, etc -- just in case they want to make a super baton that shines like the sun or something.

**Comment by WinterDarkraven** [comment]
The light has been made softer. Below is the baton on and off.

Off:
image


On:
image

**Comment by Colovorat** [comment]
cover someone in oil
stun baton them
????
:tf:

**Comment by SomeoneYouProbablyKnow** [comment]
> cover someone in oil 
stun baton them 
???? 
:tf:

OR
Stun baton them
???
Win

**Comment by WinterDarkraven** [comment]
> cover someone in oil stun baton them ???? :tf:

I decided to test this, the baton and sparks both don't let the person holding the baton or the victim on fire, even if both are covered in oil or alcohol.

**Comment by Loafin34** [comment]
I don't like the sparks from the baton. Stunprod's fine, as it's made from an igniter that makes sparks, but a baton's supposed to be a premium item that Nanotrasen makes. It seems out of character for them to make a stun baton that's inefficient like that, sparking. I like the glow, the glow's awesome, but take out the sparks, please.

**Comment by deathrobotpunch** [comment]
I like the sparks from turning it on and off. whenever it strikes someone is a bit much.

**Comment by WinterDarkraven** [comment]
New changes.

The stun baton now sparks when activated, not when attacking.
Stun prod now actually sparks on every stun attack, instead of it being when the sparker was activated in the old code. 
Stun baton and stun prod glows
Stun baton now sparks and flickers its light rapidly when exposed to EMPs

---

I still don't know if removing the sparker code from the stun prod was, uh, good or not -- so I added it back, since the stuff I added isn't needed anymore. I'd rather not touch code if I can, since I'm not too comfortable coding. 

Below is what happens when you get EMPed now:

https://github.com/tgstation/tgstation/assets/70779744/6a4054bd-167c-47c8-ac13-a0699965d8f9

**Comment by WinterDarkraven** [comment]
No more changes are planned for this, unless anything is requested or fixes are required. Everything looks like it works!

**Review by Ghommie — APPROVED** [review]

**Comment by EricZilla** [comment]
Ironically this is a pretty good balance PR since you can more easily tell if someone is about to harm batong you

**Pull request merged**